src/twisted/internet/test/test_asyncioreactor.py

```diff
@@ -5,35 +5,43 @@
 Tests for L{twisted.internet.asyncioreactor}.
 """
 import gc
+import sys
 from unittest import skipIf
 
+from twisted.python.runtime import platform
 from twisted.trial.unittest import SynchronousTestCase
 from .reactormixins import ReactorBuilder
 
+from twisted.internet.asyncioreactor import AsyncioSelectorReactor
+from asyncio import (
+    set_event_loop, set_event_loop_policy,
+    DefaultEventLoopPolicy, Future, SelectorEventLoop)
+
+hasWindowsProactorEventLoopPolicy = False
+hasWindowsSelectorEventLoopPolicy = False
+
 try:
-    from twisted.internet.asyncioreactor import AsyncioSelectorReactor
-    import asyncio
+    if sys.platform.startswith("win32"):
+        from asyncio import (
+            WindowsProactorEventLoopPolicy, WindowsSelectorEventLoopPolicy)
+        hasWindowsProactorEventLoopPolicy = True
+        hasWindowsSelectorEventLoopPolicy = True
 except ImportError:
-    doSkip = True
-else:
-    doSkip = False
+    pass
 
 
 
-@skipIf(doSkip, "Requires asyncio.")
 class AsyncioSelectorReactorTests(ReactorBuilder, SynchronousTestCase):
     """
     L{AsyncioSelectorReactor} tests.
     """
-    def test_defaultEventLoopFromGlobalPolicy(self):
-        """
-        L{AsyncioSelectorReactor} wraps the global policy's event loop
-        by default.  This ensures that L{asyncio.Future}s and
-        coroutines created by library code that uses
-        L{asyncio.get_event_loop} are bound to the same loop.
-        """
-        reactor = AsyncioSelectorReactor()
-        future = asyncio.Future()
+
+    _defaultEventLoop = DefaultEventLoopPolicy().new_event_loop()
+    _defaultEventLoopIsSelector = isinstance(_defaultEventLoop,
+                                             SelectorEventLoop)
+
+    def _test_reactor(self, reactor):
+        future = Future()
```

Comment:
I think that this needs a docstring.
It can be renamed to something like `def assertSharedEventLoop` .
No need to make it private as this is testing code.
To me, this looks like specialized assertion code.

Reply:
Done